### PR TITLE
params diff: skip params dependency for vars

### PR DESCRIPTION
**Pull request by skshetry**
Previously, `configs` which was a list of PathInfo was being used
to check if the `vars` should be skipped or not.

But, the file is stored as a string that made that check redundant
which might have caused #5117 as they were being stored twice,
once as an expanded paramdependency and next as a flattened dict
from vars, which later during `flatten`, it might have panicked
from that duplication.

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/params/show.py [hunk 1]
```diff
@@ -50,17 +51,18 @@ def _read_params(repo, configs, rev):
     return res
 
 
-def _add_vars(repo, configs, params):
+def _add_vars(repo, params):
```
Comment (pared) [inline dvc/repo/params/show.py:54]: I think that maybe we should change the naming, too. Now we are not adding anything. How about `_stage_vars`, to stress the fact that those are params loaded from the stage?

**Review by pared — APPROVED** [review]

File: dvc/repo/params/show.py (hunk 1 above)
Comment (skshetry) [inline dvc/repo/params/show.py:54]:
```suggestion
def _collect_vars(repo, params):
```

File: dvc/repo/params/show.py [hunk 2]
```diff
@@ -70,7 +72,11 @@ def show(repo, revs=None):
     for branch in repo.brancher(revs=revs):
         configs = _collect_configs(repo, branch)
         params = _read_params(repo, configs, branch)
-        _add_vars(repo, configs, params)
+        vars_params = _add_vars(repo, params)
```
Comment (skshetry) [inline dvc/repo/params/show.py:75]:
```suggestion
        vars_params = _collect_vars(repo, params)
```

**Review by pmrowla — APPROVED** [review]

**Comment by skshetry** [comment]
@Mergifyio backport 1.11

**Pull request merged**